cmd/plugin.go

```diff
@@ -582,9 +604,62 @@ func runPluginListCmd(cmd *cobra.Command, args []string) {
 		return
 	}
 
+	err := showPluginListOutput(pluginList, failedPluginMap, missingPluginMap, res, outputFormat)
+	if err != nil {
+		error_helpers.ShowError(cmd.Context(), err)
+	}
+
+}
+
+func getPluginList(ctx context.Context) (pluginList []plugin.PluginListItem, failedPluginMap, missingPluginMap map[string][]*modconfig.Connection, res *modconfig.ErrorAndWarnings) {
+	statushooks.Show(ctx)
+	defer statushooks.Done(ctx)
+
+	// get the maps of available and failed/missing plugins
+	pluginConnectionMap, failedPluginMap, missingPluginMap, res := getPluginConnectionMap(ctx)
+	if res.Error != nil {
+		return nil, nil, nil, res
+	}
+
+	// TODO do we really need to look at installed plugins - can't we just use the plugin connection map
+	// get a list of the installed plugins by inspecting the install location
+	// pass pluginConnectionMap so we can populate the connections for each plugin
+	pluginList, err := plugin.List(pluginConnectionMap)
+	if err != nil {
+		res.Error = err
+		return nil, nil, nil, res
+	}
+
+	// remove the failed plugins from `list` since we don't want them in the installed table
+	for pluginName := range failedPluginMap {
+		for i := 0; i < len(pluginList); i++ {
+			if pluginList[i].Name == pluginName {
+				pluginList = append(pluginList[:i], pluginList[i+1:]...)
+				i-- // Decrement the loop index since we just removed an element
+			}
+		}
+	}
+	return pluginList, failedPluginMap, missingPluginMap, res
+}
+
+func showPluginListOutput(pluginList []plugin.PluginListItem, failedPluginMap, missingPluginMap map[string][]*modconfig.Connection, res *modconfig.ErrorAndWarnings, outputFormat string) error {
+	if outputFormat == "table" {
+		showPluginListAsTable(pluginList, failedPluginMap, missingPluginMap)
+	} else if outputFormat == "json" {
+		return showPluginListAsJSON(pluginList, failedPluginMap, missingPluginMap)
+	}
+	if len(res.Warnings) > 0 {
+		fmt.Println()
+		res.ShowWarnings()
+		fmt.Printf("\n")
+	}
+	return nil
+}
+
+func showPluginListAsTable(pluginList []plugin.PluginListItem, failedPluginMap, missingPluginMap map[string][]*modconfig.Connection) {
```

Comment:
```suggestion
func showPluginListAsTable(pluginList []plugin.PluginListItem, failedPluginMap, missingPluginMap map[string][]*modconfig.Connection, res *modconfig.ErrorAndWarnings) error {
```

Take `res` as an argument to this func, and also as an argument to the  showPluginListAsJSON func(for consistency)
This func can be modified into:
```
func showPluginListAsTable(pluginList []plugin.PluginListItem, failedPluginMap, missingPluginMap map[string][]*modconfig.Connection, res *modconfig.ErrorAndWarnings) error {
	// List installed plugins in a table
	if len(pluginList) != 0 {
		headers := []string{"Installed", "Version", "Connections"}
		var rows [][]string
		for _, item := range pluginList {
			rows = append(rows, []string{item.Name, item.Version, strings.Join(item.Connections, ",")})
		}
		display.ShowWrappedTable(headers, rows, &display.ShowWrappedTableOptions{AutoMerge: false})
		fmt.Printf("\n")
	}

	// List failed/missing plugins in a separate table
	if len(failedPluginMap)+len(missingPluginMap) != 0 {
		headers := []string{"Failed", "Connections", "Reason"}
		var conns []string
		var missingRows [][]string

		// failed plugins
		for p, item := range failedPluginMap {
			for _, conn := range item {
				conns = append(conns, conn.Name)
			}
			missingRows = append(missingRows, []string{p, strings.Join(conns, ","), constants.ConnectionErrorPluginFailedToStart})
			conns = []string{}
		}

		// missing plugins
		for p, item := range missingPluginMap {
			for _, conn := range item {
				conns = append(conns, conn.Name)
			}
			missingRows = append(missingRows, []string{p, strings.Join(conns, ","), constants.InstallMessagePluginNotInstalled})
			conns = []string{}
		}

		display.ShowWrappedTable(headers, missingRows, &display.ShowWrappedTableOptions{AutoMerge: false})
		fmt.Println()
	}

	if len(res.Warnings) > 0 {
		fmt.Println()
		res.ShowWarnings()
		fmt.Printf("\n")
	}
	return nil
}
```